pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/SchedulerManager.java

```diff
@@ -241,6 +252,12 @@ private void publishNewAssignment(Assignment assignment, boolean deleted) {
         for (Map.Entry<String, Function.Instance> instanceEntry : functionInstances.entrySet()) {
             String fullyQualifiedInstanceId = instanceEntry.getKey();
             Function.Instance instance = instanceEntry.getValue();
+            String heartBeatWorkerId = checkHeartBeatFunction(instance);
+            if (heartBeatWorkerId != null) {
+                resultAssignments
```

Comment:
you can probably just exclude/skip the heartbeat function from instead of using "resultAssignments"